offchain/eth-input-reader/src/eth_input_reader.rs

```diff
@@ -49,10 +58,23 @@ pub async fn start(
             .await
             .context(BrokerSnafu)?;
 
+    trace!("Creating env");
+    let env = create_env(
```

Comment:
I think the granularity of the setup function should be larger.

I'd create a function that returns both the `env` and the `block_subscriber`. The reason is because you don't need the `provider` nor the `subscriber`. Or at least do something like:

```
let (subscription, env) = {
   let provider = ...;
   let subscriber = ...;
};
```